### PR TITLE
Add support for STM32Duino bootloaders on macOS

### PR DESCRIPTION
**Pull request by jhawthorn**
Similar to #166, but for macOS 🍏.

## Description

Adds support for the STM32duino bootloader, VID=0x1eaf, PID=0x0003. A number of keyboards in qmk_firmware already target this style of bootloader. It's very convenient on STM32f1 devices, which don't ship with their own or DIY keyboards based on the popular bluepill breakout.

Bootloader is https://github.com/rogerclarkmelbourne/STM32duino-bootloader which also has a few forks.

Screen Shot 2020-09-22 at 19 04 45

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Documentation

### CODE REVIEW

**Comment by jhawthorn** [comment]
If it helps to get this merged I'm happy to rebase this onto #166 and address the feedback there 🙂

**Comment by fauxpark** [comment]
@tzarc ?

**Comment by tzarc** [comment]
Totally didn't see this until now. If you want to get this in, if it's own accord, feel free. I guess we can revisit the Windows side of things too.

**Comment by fauxpark** [comment]
I am now in possession of a Bluepill, and my ST-Link should be arriving sometime this week, so I'll be able to test out both of these PRs soon.

**Comment by tzarc** [comment]
#166 is now in.

File: osx/qmk_toolbox/Flashing.m [hunk 1]
```diff
@@ -170,6 +172,14 @@ - (void)flashKiibohdWithFile:(NSString *)file {
     }
 }
 
+- (void)flashSTM32DuinoWithFile:(NSString *)file {
+    if([[[file pathExtension] lowercaseString] isEqualToString:@"bin"]) {
+        [self runProcess:@"dfu-util" withArgs:@[@"-a", @"2", @"-d", @"1eaf:0003", @"-R", @"-D", file]];
```
Comment (fauxpark) [inline osx/qmk_toolbox/Flashing.m:177]:
```suggestion
        [self runProcess:@"dfu-util" withArgs:@[@"-a", @"2", @"-d", @"1EAF:0003", @"-R", @"-D", file]];
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by fauxpark** [comment]
Thanks!

**Pull request merged**